lib/mapObjects/CArmedInstance.cpp

```diff
@@ -89,19 +89,24 @@ void CArmedInstance::updateMoraleBonusFromArmy()
 			factionsInArmy -= mixableFactions - 1;
 	}
 
+	std::string description;
```

Comment:
Avoid changing shared variable. b->description directly. Not sure why boost::algorithm::trim(b->description) crashes in multithreading environment but it does.